rslib/src/search/sqlwriter.rs

```diff
@@ -598,25 +766,47 @@ impl SqlWriter<'_> {
         Ok(())
     }
 
-    fn write_regex(&mut self, word: &str) {
-        self.sql.push_str("n.flds regexp ?");
+    fn write_regex(&mut self, word: &str, no_combining: bool) -> Result<()> {
+        let flds_expr = if no_combining {
+            "coalesce(without_combining(n.flds), n.flds)"
+        } else {
+            "n.flds"
+        };
+        let word = if no_combining {
+            without_combining(word)
+        } else {
+            std::borrow::Cow::Borrowed(word)
+        };
         self.args.push(format!(r"(?i){}", word));
-    }
+        let arg_idx = self.args.len();
+        if let Some(field_indices_by_notetype) = self.excluded_fields_by_notetype()? {
+            let notetype_clause = |UnqualifiedRegexSearchContext{mid, fields}: &UnqualifiedRegexSearchContext| -> String {
+                let clause = if fields.is_empty() {
+                    format!("{flds_expr} regexp ?{arg_idx}")
+                } else {
+                    let indices = fields.iter().join(",");
+                    format!("exclude_fields({flds_expr}, {indices}) regexp ?{arg_idx}")
```

Comment:
This calls two Rust functions. Wouldn't it make sense to use the existing `regexp_fields` function instead?

Reply:
Makes sense. It's even 3 functions in the no_combining case.